fuel-tx/src/transaction/types/script.rs

```diff
@@ -102,7 +103,12 @@ impl crate::UniqueIdentifier for Script {
         clone.outputs_mut().iter_mut().for_each(Output::prepare_sign);
         clone.witnesses_mut().clear();
 
-        fuel_crypto::Hasher::hash(clone.to_bytes().as_slice())
+        let mut hasher = Hasher::default();
+        // chain ID
+        hasher.input(params.chain_id.to_be_bytes());
+        // transaction bytes
+        hasher.input(clone.to_bytes().as_slice());
+        hasher.finalize()
```

Comment:
[nit] It Would be nice to move this code in the `compute_tx_id` function and reuse in 3 places=)